clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/view/ChangeCaptureViewWriter.java

```diff
@@ -0,0 +1,196 @@
+
+    // Only the version we're transiting FROM needs to populate the topic switch message into the change capture topic
+    if (Version.parseVersionFromVersionTopicName(versionSwapMessage.oldServingVersionTopic.toString()) != version) {
+      return;
+    }
+
+    Map<String, Long> sortedWaterMarkOffsets = partitionConsumptionState.getLatestProcessedUpstreamRTOffsetMap();
+    List<Long> highWaterMarkOffsets = new ArrayList<>();
+    for (String url: sortedWaterMarkOffsets.keySet()) {
+      highWaterMarkOffsets.add(
+          kafkaClusterUrlToIdMap.getInt(url),
+          partitionConsumptionState.getLatestProcessedUpstreamRTOffsetMap().get(url));
+    }
+
+    // Write the message on veniceWriter to the change capture topic
+    if (veniceWriter == null) {
+      initializeVeniceWriter(version);
+    }
+
+    veniceWriter.sendControlMessage(
+        constructVersionSwapControlMessage(versionSwapMessage, highWaterMarkOffsets),
+        partitionConsumptionState.getPartition(),
+        Collections.emptyMap(),
+        null,
+        DEFAULT_LEADER_METADATA_WRAPPER);
+  }
+
+  @Override
+  public Map<String, VeniceProperties> getTopicNamesAndConfigsForVersion(int version) {
+    return internalView.getTopicNamesAndConfigsForVersion(version);
+  }
+
+  @Override
+  public String getWriterClassName() {
+    return internalView.getWriterClassName();
+  }
+
+  @Override
+  public void close() {
+    internalView.close();
+    veniceWriter.close();
+  }
+
+  synchronized private void initializeVeniceWriter(int version) {
+    if (veniceWriter != null) {
+      return;
+    }
+    String changeCaptureTopicName = this.getTopicNamesAndConfigsForVersion(version).keySet().stream().findAny().get();
+
+    // Build key/value Serializers for the kafka producer
+    VeniceWriterOptions.Builder configBuilder = new VeniceWriterOptions.Builder(changeCaptureTopicName);
+    VeniceKafkaSerializer keySerializer = new VeniceAvroKafkaSerializer(keySchema);
+    VeniceKafkaSerializer valueSerializer = new VeniceAvroKafkaSerializer(RecordChangeEvent.getClassSchema());
+    configBuilder.setKeySerializer(keySerializer);
+    configBuilder.setValueSerializer(valueSerializer);
+
+    // Set writer properties based on the store version config
+    Version storeVersionConfig = store.getVersion(version).get();
+    PartitionerConfig partitionerConfig = storeVersionConfig.getPartitionerConfig();
+    if (partitionerConfig != null) {
+      // TODO: It would make sense to give the option to set a different partitioner for this view. Might
+      // want to consider adding it as a param available to this view type.
+      VenicePartitioner venicePartitioner = PartitionUtils.getVenicePartitioner(partitionerConfig);
+      configBuilder.setPartitioner(venicePartitioner);
+    }
+    configBuilder.setChunkingEnabled(storeVersionConfig.isChunkingEnabled());
+    veniceWriter = new VeniceWriterFactory(props).createVeniceWriter(configBuilder.build());
```

Comment:
As we create one `veniceWriter` per store version, or maybe just have one more `veniceWriter` per store. I remembered we previouslly having issues with too many producers (producer has a cost of buffer of around 50MB.) causing OOM issue, We do not have shared producers now, or might not be in the future, we do not have plan to enable this feature.  Not sure if we can reuse the producer inside the `veniceWriter` for the AA RT ingestion, I think we might be careful here.